lib/worker.coffee

```diff
@@ -44,3 +48,21 @@ module.exports = (text, attributes, options) ->
       stack: stack
 
   callback()
+
+registerBlocksPositions = (blocks, result, index) ->
+  for block in blocks
+    lineno = block.$lineno()
+    if typeof lineno is 'number'
```

Comment:
`lineno` can be a number or `Opal.nil` (and I don't want to use Opal global variable)